addons/website/static/src/builder/plugins/layout_option/layout_option.js

```diff
@@ -14,6 +14,23 @@ export class LayoutOption extends BaseOptionComponent {
         SpacingOption,
         AddElementOption,
     };
+
+    setup() {
+        super.setup();
+        this.upgradeContainers();
+    }
+
+    // TODO: Remove once data-vxml="number" snippet comparison is restored.
+    upgradeContainers() {
+        // Upgrade legacy snippet for visual consistency on new versions.
+        const snippetContainerEl = this.document.querySelector(
+            ".s_kickoff .container:not([data-vxml])"
+        );
+        if (snippetContainerEl) {
+            snippetContainerEl.classList.add("s_allow_columns");
+            snippetContainerEl.dataset.vxml = "001";
+        }
+    }
```

Comment:
In my opinion, it is bad practice to use component assembly to modify the website. It creates flying mutations that are not linked to any step. The second effect is that when the user selects a block, the block rendering is impacted. In terms of UX, it doesn't look great.

Shouldn't we migrate all the blocks that have this issue when we open the editor? (Use the setup in a plugin)

Reply:
@shsa-odoo Do we have task for Snippet version comparison, if yes then we can put this task in block stage and put dependency of that task.